### ISSUE TITLE
Unit test is failing

### ISSUE THREAD

**Issue by CoderPaulK** [derjust/spring-data-dynamodb#157]
## Expected Behavior
`mvn clean package` should result in build, unit tests should run and pass

## Actual Behavior
Build fails with following error:
```
[ERROR] /Users/blah/REPOS/spring-data-dynamodb/src/test/java/org/socialsignin/spring/data/dynamodb/repository/query/AbstractDynamoDBQueryCriteriaUnitTest.java:[56,25] method addAttributeValue in class org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCriteria<T,ID> cannot be applied to given types;
  required: java.util.List<com.amazonaws.services.dynamodbv2.model.AttributeValue>,java.lang.Object,java.lang.Class<P>,boolean
  found: java.util.List<com.amazonaws.services.dynamodbv2.model.AttributeValue>,java.util.Date,java.lang.String,java.lang.Class<java.util.Date>,boolean
  reason: cannot infer type-variable(s) P
    (actual and formal argument lists differ in length)
[ERROR] /Users/blah/REPOS/spring-data-dynamodb/src/test/java/org/socialsignin/spring/data/dynamodb/repository/query/AbstractDynamoDBQueryCriteriaUnitTest.java:[92,25] method addAttributeValue in class org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCriteria<T,ID> cannot be applied to given types;
  required: java.util.List<com.amazonaws.services.dynamodbv2.model.AttributeValue>,java.lang.Object,java.lang.Class<P>,boolean
  found: java.util.List<com.amazonaws.services.dynamodbv2.model.AttributeValue>,java.util.List<java.util.Date>,java.lang.String,java.lang.Class<java.util.Date>,boolean
  reason: cannot infer type-variable(s) P
    (actual and formal argument lists differ in length)
```
=======================

## Steps to Reproduce the Problem

  1. clone repo on Mac
  2. `mvn clean package`

## Specifications
  
  - Spring Data DynamoDB Version: 
  - Spring Data Version:
  - AWS SDK Version:
  - Java Version: 
  - Platform Details: 

All those information are logged by `org.socialsignin.spring.data.dynamodb.repository.support.DynamoDBRepositoryFactory` on `INFO` level on startup.
Or use `java -version` and `mvn dependency:tree | grep -E 'spring|aws'` to provide those version numbers.

**Comment by derjust** [comment]
Thanks for the report. Saw that on the weekend, too.
Fix done, need to push

**Comment by CoderPaulK** [comment]
I have created PR for this as well

**Comment by derjust** [comment]
Thanks!
Merged via #158 / #159
